core/txpool/blobpool/blobpool.go

```diff
@@ -0,0 +1,1488 @@
+		}
+		if len(txs) == 0 {
+			delete(p.index, addr)
+			delete(p.spent, addr)
+			if inclusions != nil { // only during reorgs will the heap will be initialized
+				heap.Remove(p.evict, p.evict.index[addr])
+			}
+			p.reserve(addr, false)
+		} else {
+			p.index[addr] = txs
+		}
+		log.Warn("Dropping overdrafted blob transactions", "from", addr, "balance", balance, "spent", spent, "drop", nonces, "ids", ids)
+		for _, id := range ids {
+			if err := p.store.Delete(id); err != nil {
+				log.Error("Failed to delete blob transaction", "from", addr, "id", id, "err", err)
+			}
+		}
+	}
+	// Included cheap transactions might have left the remaining ones better from
+	// an eviction point, fix any potential issues in the heap.
+	if _, ok := p.index[addr]; ok && inclusions != nil {
+		heap.Fix(p.evict, p.evict.index[addr])
+	}
+}
+
+// offload removes a tracked blob transaction from the pool and moves it into the
+// limbo for tracking until finality.
+//
+// The method may log errors for various unexpcted scenarios but will not return
+// any of it since there's no clear error case. Some errors may be due to coding
+// issues, others caused by signers mining MEV stuff or swapping transactions. In
+// all cases, the pool needs to continue operating.
+func (p *BlobPool) offload(addr common.Address, nonce uint64, id uint64, inclusions map[common.Hash]uint64) {
+	data, err := p.store.Get(id)
+	if err != nil {
+		log.Error("Blobs missing for included transaction", "from", addr, "nonce", nonce, "id", id, "err", err)
+		return
+	}
+	item := new(blobTx)
+	if err = rlp.DecodeBytes(data, item); err != nil {
+		log.Error("Blobs corrupted for included transaction", "from", addr, "nonce", nonce, "id", id, "err", err)
+		return
+	}
+	block, ok := inclusions[item.Tx.Hash()]
+	if !ok {
+		log.Warn("Blob transaction swapped out by signer", "from", addr, "nonce", nonce, "id", id)
+		return
+	}
+	if err := p.limbo.push(item.Tx.Hash(), block, item.Blobs, item.Commits, item.Proofs); err != nil {
+		log.Warn("Failed to offload blob tx into limbo", "err", err)
+		return
+	}
+}
+
+// Reset implements txpool.SubPool, allowing the blob pool's internal state to be
+// kept in sync with the main transacion pool's internal state.
+func (p *BlobPool) Reset(oldHead, newHead *types.Header) {
+	waitStart := time.Now()
+	p.lock.Lock()
+	resetwaitHist.Update(time.Since(waitStart).Nanoseconds())
+	defer p.lock.Unlock()
+
+	defer func(start time.Time) {
+		resettimeHist.Update(time.Since(start).Nanoseconds())
+	}(time.Now())
+
+	statedb, err := p.chain.StateAt(newHead.Root)
+	if err != nil {
+		log.Error("Failed to reset blobpool state", "err", err)
+		return
+	}
+	p.head = newHead
+	p.state = statedb
+
+	// Run the reorg between the old and new head and figure out which accounts
+	// need to be rechecked and which transactions need to be readded
+	if reinject, inclusions := p.reorg(oldHead, newHead); reinject != nil {
+		for addr, txs := range reinject {
```

Comment:
The logic is a bit unintuitive. Theoretically, whenever a a new block is generated, we should recheck the "relevant" accounts, by dropping the stale transactions.

Here, the `reinject` will include all the accounts with transactions included in the block(they are not "reinjected", but the accounts who sent transactions in the latest block).

Reply:
When a new block is added but there's no reorg, the reinject map will contain an empty entry for each account that had txs in the block. As such, each account will be rechecked to invalidate blob txs, but no tx will be pushed into the pool since the content is empty, we just track that they were touched.

Reply:
I know that reject will include all accounts have txs included in the block.
I just want to say `reinject` is confusing. `reinject` feels we do the reorg
and want to reinsert a part of reorg'd transactions into pool for future packing

Reply:
I am fine with the code, since it's logically correct. I just think the var name is confusing.